aws/resource_aws_inspector_assessment_template.go

```diff
@@ -71,8 +76,8 @@ func resourceAwsInspectorAssessmentTemplateCreate(d *schema.ResourceData, meta i
 
 	d.SetId(aws.StringValue(resp.AssessmentTemplateArn))
 
-	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
-		if err := keyvaluetags.InspectorUpdateTags(conn, d.Id(), nil, v); err != nil {
+	if len(tags) > 0 {
+		if err := keyvaluetags.InspectorUpdateTags(conn, d.Id(), nil, tags); err != nil {
```

Comment:
i _think_ in this case `tags` needs to be passed in as a map or else when `New` is called on it (when inside `InspectorUpdateTags`) , it'll be a new `KeyValueTags` with the values lost

Reply:
Submitted https://github.com/hashicorp/terraform-provider-aws/pull/19011 to cover this nonintuitive situation 👍